monitoring.adoc

```diff
@@ -777,17 +777,83 @@ account activity for one user without filling up mailbox.log with log
 messages for unrelated accounts.  See <<command_line_utilities,Command-Line
 Utilities>>, the `zmprov` miscellaneous section.
 
-*Reading records in the log* The example below is a record showing that
-on June 25, 2007, the zimbra server with an IP address of 127.0.0.1 was in
-the process of deleting backups that were created on Monday, June 18, 2007
-at 8 seconds after midnight Pacific Daylight Time (PDT) or older than that
-date.
+*Log pattern*
 
-image:images/mailboxlogdescription.png[Mailbox Log Entry]
+by default log entries in `mailbox.log` have the following Log4j pattern:
+====
+%d %-5p [%t] [%z] %c{1} - %m%n
+====
+This pattern consists of 6 blocks of data:
 
-[NOTE]
-*Component thread number* identifies which thread managed by mailboxd is
-performing the action logged.
+* Date and time (e.g.: `2018-01-22 19:23:07,100`)
+* Log level (e.g. `INFO`)
+* Thread name (e.g. `[qtp1043351526-547:https:https://localhost:7071/service/admin/soap/DeleteAccountRequest]`, `[Index-9]`, etc)
+* {product-name} context
+* Component name (e.g. `soap`, `mailbox`, `mbxmgr`, etc)
+* Log message. *Note:* the log message section may span multiple lines. When a log message contains an exception, the stack trace will always start on a new line below the error message.
+
+You can read more about Log4j patterns in https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/PatternLayout.html[Log4j PatternLayout documentation].
+
+*Thread name in mailbox.log*
+
+Thread names in mailbox.log are prefixed to identify internal components. Most threads have one of the following naming conventsion: "*{thread prefix}-{thread number}*" or "*{thread prefix}-{thread number}:{url}*".
+
+The following *{thread prefix}* values are currently used for thread names in {product-name}: `btpool`, `pool`, `LmtpServer`, `ImapServer`,`ImapSSLServer`, `Pop3Server`, `Pop3SSLServer`, `ScheduledTask`, `Timer`, `AnonymousIoService`, `CloudRoutingReaderThread`, `GC`, `SocketAcceptor`, `Thread`, `qtp`.
+
+Threads with prefix `qtp` are created by Jetty `QueuedThreadPool` and have the following naming convention: "*qtp{hash code}-{thread number}:{url}*" where *{hash code}*` is the hash code value of the instance of `QueuedThreadPool` that owns the thread (see https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#hashCode()[Object::hashCode] in Java platform documentation).
+
+*{thread number}* in thread names is an integer that monotonously increases within each thread factory. Thread numbers are reset when `mailboxd` process is stopped or restarted.
```

Comment:
s/monotonously/monotonically/  - although I agree that it is a bit monotonous how those integers just keep increasing.